app/models/feature_completion.rb

```diff
@@ -0,0 +1,9 @@
+class FeatureCompletion < ActiveRecord::Base
+  belongs_to :split
+  belongs_to :app
+
+  validates :split, uniqueness: { scope: :app }
+  validates :split, :app, :version, presence: true
+
+  attribute :version, :app_version
```

Comment:
I used the Rails 5 Attributes API to make this type conversion work. It's great and will come in handy for auto-casting string representations of version numbers coming in via RESTful APIs later.